pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -106,19 +105,16 @@ def advance(  # type: ignore[override]
         void(dl_max_batches)
 
         assert self._dataloader_iter is not None
-        batch_idx, (batch, self.batch_progress.is_last_batch) = next(self._dataloader_iter)
-
+        batch, self.batch_progress.is_last_batch = next(self._dataloader_iter)
         if batch is None:
             raise StopIteration
 
-        if not data_fetcher.store_on_device:
-            batch = self.trainer._call_strategy_hook("batch_to_device", batch, dataloader_idx=(dataloader_idx or 0))
+        # configure step_kwargs
+        # TODO: each loop should construct its own kwargs, so we avoid the dataloader_idx reference here
+        kwargs = self._build_kwargs(batch, self.batch_progress.current.ready, dataloader_idx)
 
         self.batch_progress.increment_ready()
```

Comment:
```suggestion
        self.batch_progress.increment_ready()

        # configure step_kwargs
        # TODO: each loop should construct its own kwargs, so we avoid the dataloader_idx reference here
        kwargs = self._build_kwargs(batch, self.batch_idx, dataloader_idx)
```
just for better code-readability. will this break anything?

Reply:
> will this break anything?

Yes. two issues there:

- EvaluationEpochLoop does not define `self.batch_idx`
- We need to `increment_ready` after accessing it.